src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -2195,6 +2195,49 @@ test_mongoc_client_unix_domain_socket (void *context)
    bson_free (uri_str);
 }
 
+static void
+test_mongoc_client_bind_ip (void)
+{
+   mongoc_client_t *client;
+   bson_error_t error;
+
+   client = mongoc_client_new ("mongodb://localhost");
+   BSON_ASSERT (client);
+   ASSERT_OR_PRINT (
+      mongoc_client_read_command_with_opts (
+         client, "admin", tmp_bson ("{'ping': 1}"), NULL, NULL, NULL, &error),
+      error);
+   mongoc_client_destroy (client);
+   fprintf (stderr, "ran sanity check ping\n");
+
+   client = mongoc_client_new ("mongodb://localhost");
+   BSON_ASSERT (client);
+
+   // Test an IPv4 address
+   ASSERT_OR_PRINT (mongoc_client_set_bind_ip (client, "127.0.0.1", &error), error);
+   //ASSERT_OR_PRINT (mongoc_client_set_bind_ip (client, "192.168.1.14", &error), error);
```

Comment:
This is my machine's current IP address, which I used for some manual testing.  TBD on a way to programmatically use an IP besides "127.0.0.1" on evergreen hosts.

Reply:
If it makes sense, we could determine the IP of the interface to bind to with bash, and pass it in as an environment variable.